torchvision/datasets/utils.py

```diff
@@ -50,7 +51,55 @@ def bar_update(count, block_size, total_size):
     return bar_update
 
 
+_HASH_FUNCTIONS: Dict[int, _hashlib.HASH] = {
+    16: hashlib.md5(),
+    20: hashlib.sha1(),
+    28: hashlib.sha224(),
+    32: hashlib.sha256(),
+    48: hashlib.sha384(),
+    64: hashlib.sha512(),
+}
+
+
+def calculate_checksum(fpath: str, algorithm: _hashlib.HASH = hashlib.md5(), chunk_size: int = 1024 * 1024) -> str:
+    """Compute the checksum of a file.
+
+    Args:
+        fpath (str): file to checksum
+        algorithm (_hashlib.HASH): hashing algorithm
+        chunk_size (int): size of each chunk to use during update
+
+    Returns:
+        (str): the calculated checksum
+    """
+    with open(fpath, 'rb') as f:
+        for chunk in iter(lambda: f.read(chunk_size), b''):
+            algorithm.update(chunk)
+    return algorithm.hexdigest()
+
+
+def verify_checksum(fpath: str, checksum: str, **kwargs: Any) -> bool:
+    """Verify the checksum of a file.
+
+    Args:
+        fpath (str): file to checksum
+        checksum (str): expected checksum
+
+    Returns:
+        (bool): True if checksums match, else False
+
+    Raises:
+        ValueError: if checksum does not match a known hashing algorithm
+    """
+    length = len(checksum)
+    if length in _HASH_FUNCTIONS:
+        return checksum == calculate_checksum(fpath, _HASH_FUNCTIONS[length], **kwargs)
+    else:
+        raise ValueError("Checksum does not match a known hashing algorithm.")
```

Comment:
maybe list the ones that are available with something like `', '.join(h.name for h in _HASH_FUNCTIONS.values())`